image/image.go

```diff
@@ -65,7 +65,14 @@ func (i *genericImage) Manifest() ([]byte, string, error) {
 			return nil, "", err
 		}
 		i.cachedManifest = m
-		if mt == "" {
+		is_reqmt := false
+		for _, reqmt := range i.requestedManifestMIMETypes {
+			if mt == reqmt {
+				is_reqmt = true
+				break
+			}
+		}
+		if !is_reqmt {
```

Comment:
(And this would definitely benefit from a comment mentioning that yes, $something_specific can happen, so that this code is not thrown away in a next refactoring.)